src/utilities/createRouter.ts

```diff
@@ -11,7 +11,7 @@ type RouterPush = (url: string, options?: { replace: boolean }) => Promise<void>
 type RouterReplace = (url: string) => Promise<void>
 
 export type Router<
-  TRoutes extends Routes
+  TRoutes extends Routes = []
```

Comment:
Added this default so that the injection key on line 26 wouldn't need the empty `[]`. All the type tests passed so think this is fine.